python-sdk/nuscenes/eval/lidarseg/render.py

```diff
@@ -0,0 +1,327 @@
+                          enumerate(self.strata_names)}
+
+        fig = plt.figure()
+        ax = fig.add_subplot(111)
+        ax.bar(list(stratified_iou.keys()),
+               list(stratified_iou.values()), color='grey')
+        ax.set_xlabel('Interval / m', fontsize=15)
+        ax.set_ylabel('mIOU', fontsize=15)
+        ax.set_ylim(top=1.1)  # Make y-axis slightly higher to accommodate tag.
+        ax.set_title('Stratified results for mIOU', fontsize=20)
+        ax.tick_params(axis='both', which='major', labelsize=15)
+
+        # Loop to add a tag to each bar.
+        for j, rect in enumerate(ax.patches):
+            ax.text(rect.get_x() + rect.get_width() / 2., rect.get_y() + rect.get_height() + 0.01,
+                    '{:.4f}'.format(list(stratified_iou.values())[j]),
+                    ha='center', va='bottom', fontsize=15)
+
+        plt.tight_layout()
+        plt.savefig(filename, dpi=dpi)
+
+        if self.verbose:
+            plt.show()
+
+    def render_stratified_per_class_metrics(self, filename: str, dpi: int = 150) -> None:
+        """
+        Renders the stratified per class metrics.
+        :param filename: Filename to save the render as.
+        :param dpi: Resolution of the output figure.
+        """
+        stratified_classes = {cls_name: [] for cls_name in self.id2name.values()}
+        for strata_metrics in self.stratified_per_class_metrics:
+            for cls, cls_iou in strata_metrics['iou_per_class'].items():
+                stratified_classes[cls].append(cls_iou)
+
+        # Delete the ignored class from the dictionary.
+        stratified_classes.pop(self.ignore_name, None)
+
+        plot_num_cols = 4
+        plot_num_rows = int(np.ceil(len(stratified_classes) / plot_num_cols))
+
+        gs = gridspec.GridSpec(plot_num_rows, plot_num_cols)
+        fig = plt.figure()
+        for n, (cls, cls_strata) in enumerate(stratified_classes.items()):
+            ax = fig.add_subplot(gs[n])
+            ax.bar(self.strata_names, cls_strata, color=np.array(self.mapper.coarse_colormap[cls]) / 255)
+            ax.set_xlabel('Interval / m', fontsize=3)
+            ax.set_ylabel('IOU', fontsize=3)
+            ax.set_ylim(top=1.1)  # Make y-axis slightly higher to accommodate tag.
+            ax.set_title('Stratified results for {}'.format(cls), fontsize=4)
+            ax.tick_params(axis='both', which='major', labelsize=3)
+
+            # Loop to add a tag to each bar.
+            for j, rect in enumerate(ax.patches):
+                ax.text(rect.get_x() + rect.get_width() / 2., rect.get_y() + rect.get_height() + 0.01,
+                        '{:.4f}'.format(cls_strata[j]),
+                        ha='center', va='bottom', fontsize=3)
+
+        plt.tight_layout()
+        plt.savefig(filename, dpi=dpi)
+
+        if self.verbose:
+            plt.show()
+
+    def get_stratified_per_class_metrics(self) -> List[Dict]:
```

Comment:
3 bins is very little. Maybe we could do 0-10, 10-20, 20-30, 30-40, 40+ ?

Reply:
Added more bins